rfc3164/machine_test.go

```diff
@@ -0,0 +1,68 @@
+package rfc3164
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/go-syslog/v2"
+	syslogtesting "github.com/influxdata/go-syslog/v2/testing"
+	"github.com/stretchr/testify/assert"
+)
+
+type testCase struct {
+	input        []byte
+	valid        bool
+	value        syslog.Message
+	errorString  string
+	partialValue syslog.Message
+}
+
+var testCases = []testCase{
+	{
+		[]byte(`<34>Jan 12 06:30:00 xxx apache: 1.2.3.4 - - [12/Jan/2011:06:29:59 +0100] "GET /foo/bar.html HTTP/1.1" 301 96 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; fr; rv:1.9.2.12) Gecko/20101026 Firefox/3.6.12 ( .NET CLR 3.5.30729)" PID 18904 Time Taken 0`),
+		true,
+		&SyslogMessage{
+			Base: syslog.Base{
+				Priority:  syslogtesting.Uint8Address(34),
+				Severity:  syslogtesting.Uint8Address(2),
+				Facility:  syslogtesting.Uint8Address(4),
+				Timestamp: syslogtesting.TimeParse(time.Stamp, "Jan 12 06:30:00"),
+				Hostname:  syslogtesting.StringAddress("xxx"),
+				Appname:   syslogtesting.StringAddress("apache"),
+				Message:   syslogtesting.StringAddress(`1.2.3.4 - - [12/Jan/2011:06:29:59 +0100] "GET /foo/bar.html HTTP/1.1" 301 96 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; fr; rv:1.9.2.12) Gecko/20101026 Firefox/3.6.12 ( .NET CLR 3.5.30729)" PID 18904 Time Taken 0`),
+			},
+		},
+		"",
+		nil,
+	},
+	// todo > other test cases pleaaaase
```

Comment:
We could add these test cases as a follow-on PRs.  What do you think?

Reply:
I'd prefer having 10 here covering most common messages. Maybe grabbing them from somewhere lese (we also have an issue here with some examples anyway).

Would you help me doing it? :)

Reply:
@leodido Yup, I'll grab log messages.  Should I just add commits to this PR for the test cases?

Reply:
Maybe we can add a bit of them and then merge it :)